### ISSUE TITLE
docker-entrypoint.sh fail startup if provided password is different than the already set password on an existing database

### ISSUE THREAD

**Issue by origliante** [docker-library/mysql#716]
So if you deploy a mysql image with the wrong volume,
but you provide a root password that is different than the existing database, 
the script just goes on, it fails in some points, you can see wrong password in the logs, but the database is started

It should NOT do that, since it might expose the wrong data, and since the only way to understand why this mixed behaviour is happening, is to look at the script and try to figure out.

To recap, I suggest:
If the root password is different than an existing DB: fail the startup

Tested on 5.7.22

**Comment by tianon** [comment]
This is unfortunately not something we can feasibly do: to even just _check_ whether the password is "different" on an already initialized database, we'd have to start up a temporarily server instance and try connecting to it (just like we do when we have to create a fresh instance), which would be *really* expensive to do on every container start. :disappointed:
